.github/workflows/gh-workflow-approve.yaml

```diff
@@ -0,0 +1,41 @@
+---
+name: Approve GitHub Workflows
+
+on:
+  pull_request_target:
+    types:
+      - labeled
+      - synchronize
+    branches:
+      - main
```

Comment:
Note: there is no `main` branch in bbolt, instead it's `master`.

Reply:
Duh, you're right.

Reply:
Perhaps in future we should align the branch to main @ahrtr?  I believe the procedure is pretty straightforward in GitHub.  It makes sense to try and use same across all sig-etcd subprojects if possible I think.

Reply:
I can leave both `main` and `master` in the meantime, let me know what you think. So, when we rename the branch to `main` there's not gonna be a dependency here.

Reply:
> Perhaps in future we should align the branch to main

OK, let me do it on weekend.

Reply:
cc @jmhbnz @serathius @wenjiaswe @fuweid @tjungblu @Elbehery 

If there is no any objection, I will rename the `master` to `main` on weekend. 

Reference: https://www.git-tower.com/learn/git/faq/git-rename-master-to-main

Reply:
You can also do it easily from the GitHub repository settings page (ref: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/changing-the-default-branch)

![Screenshot from 2023-12-08 12-09-56(1)](https://github.com/etcd-io/bbolt/assets/27286/4a103289-4dc6-4967-b2d4-60d7bcd5b7a8)
![Screenshot from 2023-12-08 12-10-11](https://github.com/etcd-io/bbolt/assets/27286/b809edab-af1f-41c6-b660-fabd99899a27)


If it's fine with you, I'll revert my last change and set the branch name to `main`, and you can merge after you have renamed the main branch :v: